aptos/contract/aux/sources/stake.move

```diff
@@ -0,0 +1,1301 @@
+/// Simple coin staking module, modeled after SushiSwap's MasterChef: https://github.com/sushiswap/sushiswap/blob/archieve/canary/contracts/MasterChef.sol
+module aux::stake {
+    use std::signer;
+    use std::table::{Table, Self};
+
+    use aptos_framework::coin::{Coin, Self};
+    use aptos_framework::timestamp;
+    use aptos_framework::event::{EventHandle, Self};
+    use aptos_framework::account;
+
+    use aux::authority;
+
+
+    /*************/
+    /* CONSTANTS */
+    /*************/
+    const MIN_DURATION_MS: u64 = 3600 * 24 * 1000000; // 1 day
+    const MAX_DURATION_MS: u64 = 3600 * 24 * 365 * 1000000; // 1 year
```

Comment:
do we need to consider leap year?

(and on that same note, leap second?)

Reply:
Don't think so on leap year, there's nothing that actually requires this to be exactly one year